clang/tools/clang-scan-deps/ClangScanDeps.cpp

```diff
@@ -744,6 +744,9 @@ getCompilationDataBase(int argc, char **argv, std::string &ErrorMessage) {
     return nullptr;
   }
 
+  // Only 1 threads is required if P1689 per file mode.
+  NumThreads = 1;
```

Comment:
Instead of setting `NumThreads` for the P1689 format, can we check if size of `Inputs` below and avoid creating the thread pool for any single-compilation database?

Reply:
Makes sense. Done.

Reply:
I still see `NumThreads = 1` here.

Reply:
Oh, sorry. I don't know why I missed this somehow. This should be done now.